src/utils.py

```diff
@@ -115,8 +115,18 @@ def install_binaries_from_urls(binary_urls: str, sha256_urls: str) -> None:
         if response.status_code != 200:
             raise ValueError(f"Download binary failed with: {response.text}. Check 'binary-url'!")
         binary_hash = hashlib.sha256(response.content).hexdigest()
-        # TODO: keeping the binary name won't work for the charm if it's not exactly 'polkadot', adjust this if more chains start using multiple binaries
-        binary_name = binary_url.split('/')[-1]
+        if 'execute-worker' in binary_url.split('/')[-1]:
+            try:
+                binary_name = c.EXECUTE_WORKER_BINARY_FILE[chain_name]
+            except KeyError:
+                binary_name = c.EXECUTE_WORKER_BINARY_FILE['default']
```

Comment:
So the intent is to actually produce a `KeyError` for all non-canary/enjin chain names?

If so perhaps leave a comment explaining that, since it's a bit of a non-standard behavior.